src/java.base/share/classes/jdk/internal/reflect/NativeMethodAccessorImpl.java

```diff
@@ -49,16 +62,27 @@ public Object invoke(Object obj, Object[] args)
         // found from the generated bytecode.
         if (++numInvocations > ReflectionFactory.inflationThreshold()
                 && !method.getDeclaringClass().isHidden()
-                && !ReflectUtil.isVMAnonymousClass(method.getDeclaringClass())) {
-            MethodAccessorImpl acc = (MethodAccessorImpl)
-                new MethodAccessorGenerator().
-                    generateMethod(method.getDeclaringClass(),
-                                   method.getName(),
-                                   method.getParameterTypes(),
-                                   method.getReturnType(),
-                                   method.getExceptionTypes(),
-                                   method.getModifiers());
-            parent.setDelegate(acc);
+                && !ReflectUtil.isVMAnonymousClass(method.getDeclaringClass())
+                && ACCESSOR_GENERATED.compareAndSet(this, false, true)) {
+            boolean succ = false;
+            try {
+                MethodAccessorImpl acc = (MethodAccessorImpl)
+                    new MethodAccessorGenerator().
+                        generateMethod(method.getDeclaringClass(),
+                                       method.getName(),
+                                       method.getParameterTypes(),
+                                       method.getReturnType(),
+                                       method.getExceptionTypes(),
+                                       method.getModifiers());
+                parent.setDelegate(acc);
+                succ = true;
+            } finally {
+                if (succ == false) {
```

Comment:
Why `succ` variable, if you can just `catch (Throwable e)` and restore the `accessorGenerated`?

Reply:
Good suggest! Will update.